src/plugins/intel_gpu/src/runtime/ocl/ocl_ext.hpp

```diff
@@ -922,6 +923,10 @@ class UsmMemory {
         }
     }
 
+    void add_offset(size_t offsetPtr) const {
+        _usm_pointer->_ptr = (void*) ((unsigned char*) _usm_pointer->_ptr + offsetPtr);
+    }
```

Comment:
Probably we shouldn't expose this function because it allows shifting the pointer of any UsmMemory. This pointer is also used for free_mem() call, and I'm not sure it will be properly handled if shifted accidently for the not-shared memory buffer. Maybe it would be better to limit shifting only with specialized c-tr like `UsmMemory(const cl::UsmHelper& usmHelper, void* usm_ptr, size_t offset = 0)`, which creates UsmHolder with `shared_memory=true` option